tests/main/cgroup-devices-v2/task.yaml

```diff
@@ -39,31 +39,17 @@ restore: |
     fi
 
 debug: |
-    if test -f "cgroup_device.progs-1" -a -f "cgroup_device.progs-2"; then
-        echo "cgroup devices before:"
-        cat "cgroup_device.progs-1"
-        echo "cgroup devices after installation:"
-        cat "cgroup_device.progs-2"
-    fi
-    if test -f "cgroup_device.progs-3"; then
-        echo "cgroup devices after connecting the interface:"
-        cat "cgroup_device.progs-3"
-    fi
-    if test -f "cgroup_device.progs-4"; then
-        echo "cgroup devices after restarting the service:"
-        cat "cgroup_device.progs-4"
-    fi
     systemctl status snap.test-snapd-service.test-snapd-service.service || true
     journalctl -u snap.test-snapd-service.test-snapd-service.service || true
 
 execute: |
     dump_cgroup_device_progs() {
-        # dump all progs but those that are assigned to systemd, in which case
-        # they have the pids list non empty
-        bpftool prog list -j | gojq -r '.[] | select(.type == "cgroup_device") | select(.pids == null) | .id'
+        # dump all snap progs
+        bpftool prog list -j | gojq -r '.[] | select(.type == "cgroup_device") | select(.name != null) | select(.name | startswith("s_")) | .name'
```

Comment:
FYI, I think all kernels where we use cgroup v2 should support attaching names to BPF programs, but let's wait for the tests to complete